### PR TITLE
gha: get ready to install genpolicy

### PR DESCRIPTION
**Pull request by danmihai1**
The changes to install and test genpolicy must come later, after CI picks up these gha changes.

Fixes: #8856

### CODE REVIEW

**Review by GabyCT — APPROVED** [review]

**Comment by wainersm** [comment]
Hi @danmihai1 !

Allow me to check if I understood. Although kata-deploy installs genpolicy tool on the cluster, for testing sake it is needed the binary on the runner machine. So it gets the same tarball used on create the kata-deploy payload image then extracts on the runner machine.

One side question: would make sense for the user to run `genpolicy` as a pod on the cluster? Likely not because as an user I wouldn't trust (on confidential terms) any binaries outside of the TCB. Anyway, I'd like to start that brainstorm.

**Comment by danmihai1** [comment]
> Hi @danmihai1 !
> 
> Allow me to check if I understood. Although kata-deploy installs genpolicy tool on the cluster, for testing sake it is needed the binary on the runner machine. So it gets the same tarball used on create the kata-deploy payload image then extracts on the runner machine.
> 
> One side question: would make sense for the user to run `genpolicy` as a pod on the cluster? Likely not because as an user I wouldn't trust (on confidential terms) any binaries outside of the TCB. Anyway, I'd like to start that brainstorm.

Indeed, a customer would run genpolicy even before they created a customer - and you described the reasons why correctly.

**Comment by fidencio** [comment]
@danmihai1, right now I see why we can have it as you proposed.  However, in the future, I'd like us to create a second tarball that can be downloaded without getting the whole kata-containers bundle.

Would it be possible to create an issue for this?

**Comment by danmihai1** [comment]
> @danmihai1, right now I see why we can have it as you proposed. However, in the future, I'd like us to create a second tarball that can be downloaded without getting the whole kata-containers bundle.
> 
> Would it be possible to create an issue for this?

Done: https://github.com/kata-containers/kata-containers/issues/8864

File: tests/integration/kubernetes/gha-run.sh [hunk 1]
```diff
@@ -236,6 +236,11 @@ function cleanup() {
 	kubectl delete -f "${tools_dir}/packaging/kata-deploy/kata-rbac/base/kata-rbac.yaml"
 }
 
+install_kata_placeholder() {
```
[the remark this answers is not in the document]
Comment (danmihai1) [reply tests/integration/kubernetes/gha-run.sh]: Great ideas to make these changes easier to understand - thanks @fidencio!

**Review by wainersm — APPROVED** [review]
With @fidencio 's suggestions it looks better now. Thanks @danmihai1 !

**Pull request merged**